src/DataWrangling/EN4/EN4_mask.jl

```diff
@@ -0,0 +1,52 @@
+using Oceananigans: location
+using Oceananigans.Architectures: AbstractArchitecture
+using Oceananigans.Grids: znode
+
+import ClimaOcean: stateindex
+
+"""
+    EN4_mask(architecture = CPU(); minimum_value = Float32(-1e5))
+
+A boolean field where `true` represents a missing value in the EN4 dataset.
+"""
+function EN4_mask(metadata, architecture = CPU();
+                   data_field = EN4_field(metadata; architecture, inpainting=nothing),
+                   minimum_value = Float32(-1e5),
+                   maximum_value = Float32(1e5))
+
+    mask  = Field{location(data_field)...}(data_field.grid, Bool)
+
+    # Set the mask with zeros where field is defined
+    launch!(architecture, data_field.grid, :xyz, _set_EN4_mask!, mask, data_field, minimum_value, maximum_value)
+
+    return mask
+end
+
+# Default
+EN4_mask(arch::AbstractArchitecture=CPU()) = EN4_mask(Metadata(:temperature, dataset=EN4Monthly()), arch)
+
+@kernel function _set_EN4_mask!(mask, Tᵢ, args...)
+    i, j, k = @index(Global, NTuple)
+    @inbounds mask[i, j, k] = (Tᵢ[i, j, k] == 1e10)
```

Comment:
This is the only line that is different from ECCO in this file - to do with the fact that missing data in EN4 is replaced with `1e10`, so we use that as the marker of land in EN4.

Reply:
looks good!

Reply:
Can we generalize this concept and define `mask_value(metadata)`, so that we only need to have one mask function?

Reply:
I've done

https://github.com/CliMA/ClimaOcean.jl/blob/b0c141c9eddeb38bc4163112675cee2251209c6f/src/DataWrangling/masking.jl#L241-L254

and defined `default_set_dataset_mask(metadata)`. But your suggestion is even better! Perhaps tomorrow I can implement it.